extensions/replication/tasks/ReplicateObject.js

```diff
@@ -139,17 +143,30 @@ class ReplicateObject extends BackbeatTask {
             sourceEntry.toFailedEntry(this.site);
         updatedSourceEntry.setReplicationSiteDataStoreVersionId(this.site,
             sourceEntry.getReplicationSiteDataStoreVersionId(this.site));
+        // TODO: understand when only metadata is updated!
+        const updateData = sourceEntry.getReplicationContent().includes('DATA');
         const kafkaEntries = [updatedSourceEntry.toKafkaEntry(this.site)];
         this.replicationStatusProducer.send(kafkaEntries, err => {
             if (err) {
-                log.error(
-                    'error in entry delivery to replication status topic',
-                    { method: 'ReplicateObject._publishReplicationStatus',
-                      topic: this.repConfig.replicationStatusTopic,
-                      entry: updatedSourceEntry.getLogInfo(),
-                      replicationStatus:
-                      updatedSourceEntry.getReplicationStatus(),
-                      error: err });
+                log.error('error in entry delivery to replication status topic', {
+                    method: 'ReplicateObject._publishReplicationStatus',
+                    topic: this.repConfig.replicationStatusTopic,
+                    entry: updatedSourceEntry.getLogInfo(),
+                    replicationStatus,
+                    error: err,
+                });
+                // TODO: Update failure metric when publish fails?
+            } else {
+                log.info('replication status published', {
+                    topic: this.repConfig.replicationStatusTopic,
+                    entry: updatedSourceEntry.getLogInfo(),
+                    replicationStatus,
+                    reason,
+                });
+                this.metricsHandler.metadataReplicationStatus({ replicationStatus });
```

Comment:
Those calls might be done as soon as entries have been sent to kafka. I'm not sure we want to take into account kafka publishing failures in those metrics.